app/models/rubygem.rb

```diff
@@ -177,7 +177,9 @@ def to_param
   end
 
   def pushable?
-    new_record? || versions.indexed.count.zero?
+    return true if new_record?
+    return false if versions.indexed.count.nonzero?
+    updated_at < 100.days.ago || created_at > 30.days.ago
```

Comment:
Damn, I missed that. sorry :(

What about something like this (this passes your tests).

```ruby
  def pushable?
    new_record? || (versions.indexed.none? && not_protected?)
  end

  def not_protected?
    updated_at < 100.days.ago || created_at > 30.days.ago
  end
```

But maybe that's bad idea. Who knows.